winlogbeat/eventlog/eventlog.go

```diff
@@ -138,3 +146,154 @@ func incrementMetric(v *expvar.Map, key interface{}) {
 		v.Add(strconv.Itoa(int(t)), 1)
 	}
 }
+
+// defaultLagPolling is the default polling period for inputMetrics.sourceLagN.
+const defaultLagPolling = time.Minute
+
+// inputMetrics handles event log metric reporting.
+type inputMetrics struct {
+	unregister func()
+	done       chan struct{}
+
+	lastBatch    time.Time
+	lastRecordID *atomic.Uint64
+
+	name        *monitoring.String // name of the provider being read
+	events      *monitoring.Uint   // total number of events received
+	dropped     *monitoring.Uint   // total number of discarded events
+	errors      *monitoring.Uint   // total number of errors
+	batchSize   metrics.Sample     // histogram of the number of events in each non-zero batch
+	sourceLag   metrics.Sample     // histogram of the difference between timestamped event's creation and reading
+	sourceLagN  metrics.Sample     // histogram of difference between the consumer's log offset and the producer's log offset
+	batchPeriod metrics.Sample     // histogram of the elapsed time between non-zero batch reads
+}
+
+// newInputMetrics returns an input metric for windows event logs. If id is empty
+// a nil inputMetric is returned. The ID delta between OS events and read events
+// will be polled each poll duration.
+func newInputMetrics(name, id string, poll time.Duration) *inputMetrics {
+	if id == "" {
+		return nil
+	}
+	reg, unreg := inputmon.NewInputRegistry(name, id, nil)
```

Comment:
Let's make the input name be a constant `winlog`. This way when viewing this in agent we have `input` values like `winlog`, `udp`, `cel`, `lumberjack`, etc.

Reply:
That should happen in the filebeat end, since this will also be being called from winlogbeat.